### ISSUE TITLE
Fatal Error loading image - unrecognized data stream contents when reading image file

### ISSUE THREAD

**Issue by UrbanLegendXV** [victorchall/EveryDream2trainer#16]
****looks like it was auto or notepad++ breaking some images during replace token process****

I did redo all of the images and it is still giving this issue. I have cleared directories and reinstalled as well.

I am [pre-processing in automatic](https://i.postimg.cc/SKRpsyng/Screenshot-20230117-060058.png) then using [BRU](https://www.bulkrenameutility.co.uk/Download.php) and [N++](https://www.ghacks.net/2016/09/16/use-notepad-to-find-text-in-all-files-of-a-folder/) to change values to match folder and tags for subjects respectively. 

It DID work with this. 

**_What automation processes regarding theses steps have you all found to be most efficient and effective?_**

It DID work with four folders of subjects the first go so I know it can. I have the model and it's children that work. I just want it to hit harder.

Great work. I'm so happy to see 8bit on windows.

01/17/2023 05:37:49 AM    ED1 mode: Overiding disable_xformers to True
01/17/2023 05:37:49 AM  Seed: 555
01/17/2023 05:37:49 AM Logging to logs\01DSU90 - GMM&co_20230117-053749
01/17/2023 05:37:49 AM  unet attention_head_dim: [8, 8, 8, 8]
01/17/2023 05:37:51 AM xformers not available or disabled
01/17/2023 05:37:51 AM  * Using FP32 *
01/17/2023 05:37:53 AM [36m * Training Text Encoder *[0m
01/17/2023 05:37:53 AM [36m * Using AdamW 8-bit Optimizer *[0m
01/17/2023 05:37:53 AM [36m * Optimizer: AdamW8bit *[0m
01/17/2023 05:37:53 AM     betas: (0.9, 0.999), epsilon: 1e-08 *[0m
01/17/2023 05:37:53 AM  * Creating new dataloader singleton
01/17/2023 05:37:53 AM * DLMA resolution 512, buckets: [[512, 512], [576, 448], [448, 576], [640, 384], [384, 640], [768, 320], [320, 768], [896, 256], [256, 896], [1024, 256], [256, 1024]]
01/17/2023 05:37:53 AM  Preloading images...
01/17/2023 05:37:53 AM  ** Trainer Set: 4, num_images: 38, batch_size: 10
01/17/2023 05:37:53 AM  Pretraining GPU Memory: 11340 / 49140 MB
01/17/2023 05:37:53 AM  saving ckpts every 1000000000.0 minutes
01/17/2023 05:37:53 AM  saving ckpts every 100 epochs
01/17/2023 05:37:53 AM  unet device: cuda:0, precision: torch.float32, training: True
01/17/2023 05:37:53 AM  text_encoder device: cuda:0, precision: torch.float32, training: True
01/17/2023 05:37:53 AM  vae device: cuda:0, precision: torch.float32, training: False
01/17/2023 05:37:53 AM  scheduler: <class 'diffusers.schedulers.scheduling_ddpm.DDPMScheduler'>
01/17/2023 05:37:53 AM  [32mProject name: [0m[92m01DSU90 - GMM&co[0m
01/17/2023 05:37:53 AM  [32mgrad_accum: [0m[92m1[0m
01/17/2023 05:37:53 AM  [32mbatch_size: [0m[92m10[0m
01/17/2023 05:37:53 AM  [32mepoch_len: [92m4[0m
01/17/2023 05:38:22 AM Fatal Error loading image: C:\Users\TemporalLabSol\Desktop\EveryDream2trainer-main\EveryDream2trainer-main\SUBJECT\01DSU90\01DSU90 3-1-318632221_8171851539551764_21628442878249915_n.png:
01/17/2023 05:38:22 AM unrecognized data stream contents when reading image file

**EL JSON**
{
  "amp": false,
  "batch_size": 10,
  "ckpt_every_n_minutes": null,
  "clip_grad_norm": null,
  "clip_skip": 0,
  "cond_dropout": 0.04,
  "data_root": "C:\\Users\\TemporalLabSol\\Desktop\\EveryDream2trainer-main\\EveryDream2trainer-main\\SUBJECT",
  "disable_textenc_training": false,
  "disable_xformers": false,
  "ed1_mode": true,
  "flip_p": 0.0,
  "gpuid": 0,
  "gradient_checkpointing": true,
  "grad_accum": 1,
  "logdir": "logs",
  "log_step": 25,
  "lowvram": false,
  "lr": 3.5e-06,
  "lr_decay_steps": 0,
  "lr_scheduler": "constant",
  "lr_warmup_steps": null,
  "max_epochs": 90,
  "notebook": false,
  "project_name": "01DSU90 - GMM&co",
  "resolution": 512,
  "resume_ckpt": "sd_v1-5_vae",
  "sample_prompts": "sample_prompts.txt",
  "sample_steps": 300,
  "save_ckpt_dir": null,
  "save_every_n_epochs": 100,
  "save_optimizer": false,
  "scale_lr": false,
  "seed": 555,
  "shuffle_tags": false,
  "useadam8bit": true,
  "wandb": false,
  "write_schedule": false,
  "rated_dataset": false,
  "rated_dataset_target_dropout_rate": 50,
  "save_full_precision": false,
  "disable_unet_training": false,
  "mixed_precision": "fp32",
  "rated_dataset_target_dropout_percent": 50,
  "config": "train.json"
}
`

**Comment by victorchall** [comment]
The file SUBJECT\01DSU90\01DSU90 3-1-318632221_8171851539551764_21628442878249915_n.png
is corrupt.  You may try opening it in any image editor program and save it again.

**Comment by victorchall** [comment]
I updated the code so it should catch these on preloading.  Not much more I can do with it, if PIL thinks it is corrupt it cannot be loaded and would need to be corrected outside the training process.
